### DISCUSSION TITLE
Static value on multi-fields relation

### DISCUSSION THREAD

**Discussion by yosietserga** [prisma/prisma#13907] [Q&A]
Hi! 

I have a simple EAV implementation with a table like this: 

```prisma
model Property {
  id          Int     @id @default(autoincrement())
  uuid        String  @unique @default(cuid())
  object_id   Int
  object_type String
  data_type   String
  group       String
  key         String
  value       String?
  status      Int?
  createdAt   DateTime @default(now())
  updatedAt       DateTime? @updatedAt

  person      Person      @relation(fields: [object_id, object_type], references: [id, "person"], onDelete: Cascade)

  @@index([object_id, object_type], name: "object_index")
  @@unique([object_id, object_type, group, key], name: "object_property")
}
```

With this single scheme I can set custom fields, but I want to know if it is possible to make a relation like this:

```prisma
person      Person      @relation(fields: [object_id, object_type], references: [id, "person"], onDelete: Cascade)
```

Where the relation has two fields, but one of those is an static appointing to the owner table:

```prisma
model Person {
  id              Int      @id @default(autoincrement())
  uuid            String   @unique @default(cuid())
  email           String   @unique
  phone           String   @unique
  password        String?
  firstname       String?
  lastname        String?
  gender          String?
  dateOfBirthday  DateTime?
  status          Int?
  deleted         Int?
  createdAt       DateTime  @default(now())
  updatedAt       DateTime?
  
  profiles        Profile[]
  accounts        Account[]
  sessions        Session[]
  properties      Property[]
}
```

The idea is that the SQL resulting looks like this:

```sql
foreign key (object_id, object_type) references Person (id, "person")
```

1. Is this possible?
2. How is the correct way to do it into the Prisma Scheme definition?

**Comment by austincrim** [comment]
Hey @yosietserga 👋,

This is not possible Prisma today. The references property has contain actual field names from your models.

Do you know if this pattern is possible in your database provider? If so, you could try generating and manually editing the migration file in order to get this functionality ([relevant docs](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/include-unsupported-database-features)).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
